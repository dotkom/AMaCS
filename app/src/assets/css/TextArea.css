.component {
  background: #F4F4F4;
  border: 1px solid #E0E0E0;
  color: #333333;
  display: block;
  width: 100%;
  min-height: 20rem;
}

.component::placeholder {
  color: #B8B8B8
}
