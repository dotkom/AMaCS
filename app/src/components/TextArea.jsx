import React from 'react';

import _s from 'assets/css/TextArea.css';

function TextArea({ text, onChange, ...rest }) {
  return (
    <textarea onChange={(e) => onChange(e.target.value)} className={_s.component} {...rest} />
  );
}

TextArea.defaultProps = {
  text: '',
  onChange: () => {}
}

export default TextArea;
