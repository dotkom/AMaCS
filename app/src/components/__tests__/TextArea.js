import React from 'react';
import { shallow } from 'enzyme';

import TextArea from '../TextArea';

describe('TextArea', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <TextArea />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with text', () => {
    const wrapper = shallow(
      <TextArea />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onChange property when text changes', () => {
    const onChangeMock = jest.fn();
    const changedText = 'Test text';
    const wrapper = shallow(
      <TextArea
      text="Test tex"
        onChange={onChangeMock}
      />
    );

    wrapper.simulate('change', { target: { value: changedText}});
    expect(onChangeMock).toHaveBeenCalledWith(changedText);
  });

  it('works without onChange ', () => {
    const onChangeMock = jest.fn();
    const changedText = 'Test text';
    const wrapper = shallow(
      <TextArea
        text="Test text"
      />
    );

    wrapper.simulate('change', { target: { value: changedText}});
    expect(wrapper).toMatchSnapshot();
  });

  it('forwards extra props', () => {
    const wrapper = shallow(
      <TextArea
        placeholder="Placeholder text"
      />
    );
    expect(wrapper).toMatchSnapshot();
  })
});
