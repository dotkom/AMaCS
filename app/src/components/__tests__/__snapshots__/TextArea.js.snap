// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea forwards extra props 1`] = `
<textarea
  className="component"
  onChange={[Function]}
  placeholder="Placeholder text"
/>
`;

exports[`TextArea renders correctly 1`] = `
<textarea
  className="component"
  onChange={[Function]}
/>
`;

exports[`TextArea renders with text 1`] = `
<textarea
  className="component"
  onChange={[Function]}
/>
`;

exports[`TextArea works without onChange  1`] = `
<textarea
  className="component"
  onChange={[Function]}
/>
`;
